src/V3Dead.cpp

```diff
@@ -542,6 +542,14 @@ void V3Dead::deadifyModules(AstNetlist* nodep) {
     V3Global::dumpCheckGlobalTree("deadModules", 0, dumpTreeLevel() >= 6);
 }
 
+void V3Dead::deadifyModuleDTypes(AstNetlist* nodep) {
+    UINFO(2, __FUNCTION__ << ": " << endl);
+    {
+        DeadVisitor{nodep, false, true, false, false, !v3Global.opt.topIfacesSupported()};
```

Comment:
Failed to garbage-collect all unnecessary types and modules together :(  Maybe can improve the algorithm first? Probably related to #2227